### PR TITLE
refactor: remove `@types/vscode` dependency

### PR DESCRIPTION
**Pull request by sapirpol**

### CODE REVIEW

**Comment by bd82** [comment]
failing build is due to coveralls permissions, ignore that.

File: packages/logger/api.d.ts [hunk 1]
```diff
@@ -4,14 +4,18 @@
...
 import { IVSCodeExtLogger, LogLevel } from "@vscode-logging/types";
-import { OutputChannel } from "vscode";
-
 export {
   IVSCodeExtLogger,
   IChildLogger,
   LogLevel
 } from "@vscode-logging/types";
 
+export type BasicOutputChannel = {
```
Comment (bd82) [inline packages/logger/api.d.ts:13]: Document that this is a **sub-type** of VSCode.OutputChannel
And that we are defining it to avoid direct dependency to `@vscode/types`
Because in some edge cases having multiple versions of `@vscode/types`
can cause compilation errors.

Comment (bd82) [reply packages/logger/api.d.ts:13]: Also add a "**compilation**" test that ensures `BasicOutputChannel` is assignable to `vscode.OutputChannel`

**Review by bd82 — CHANGES_REQUESTED** [review]
see inline comments

**Review by bd82 — APPROVED** [review]

**Review by saggir — APPROVED** [review]
